fcrepo-exts/fcrepo-migration-validator
# Update Migration Validator for Migration Utils Changes

**JIRA Ticket**: https://fedora-repository.atlassian.net/browse/FCREPO-3772

# What does this Pull Request do?

Updates the Migration Validator to account for information in RELS-INT which can result in additional versions when running the migreation utils.

# What's new?

* Before validating objects, process RELS-INT to look for downloadFilename prop
* Update how we compute the number of expected versions to include those in RELS-INT
* Add RELS-INT entry to VersionValidationIT F3 test data
* Regenerate VersionValidationIT F6 OCFL test data
* Update VersionValidationIT to check for correct number of RELS-INT versions
* Fix RDFs use in ObjectValidationIT

# How should this be tested?

* Compile + run it tests: `mvn clean verify`
* Run against the brown dataset (or any other available): ```java -jar target/fcrepo-migration-validator-1.1.0-SNAPSHOT-driver.jar -r /data/migration/results -c /data/migration/brown-f6/data/ocfl-root -s legacy -d /data/migration/brown/repoarchive/datastreams_2019 -o /data/migration/brown/repostore/data_2019/objects```

# Additional Notes:

I pulled out one of the objects in the brown dataset to test longer chains of adding/deleting filenames from RELS-INT, but unfortunately removed it while generating the IT test data. it might be good to do something like that again before merging as a sanity check. 

# Interested parties
Tag (@ mention) interested parties or, if unsure, @fcrepo/committers
